client_with_ssl.go

```diff
@@ -57,371 +56,136 @@ func main() {
-	sql := "SELECT c.country from country c"
-	fmt.Println("Select all countries with sql = " + sql)
-	result, err := sqlService.Execute(ctx, sql)
-	if err != nil {
-		fmt.Errorf("querying: %w", err)
-	}
-	defer result.Close()
-	iter, err := result.Iterator()
-	if err != nil {
-		fmt.Errorf("acquiring iterator: %w", err)
-	}
-	for iter.HasNext() {
-		row, err := iter.Next()
-		if err != nil {
-			fmt.Errorf("iterating: %w", err)
-		}
-		country := mustGet(row.GetByColumnName("country"))
-		fmt.Printf("Country = %s\n", country)
-	}
-	fmt.Println("--------------------")
+type City struct {
+	Country    string
+	CityName   string
+	Population int32
 }
 
-func createMappingForCities(sqlService sql.Service) {
-	// see: https://docs.hazelcast.com/hazelcast/5.1/sql/mapping-to-maps#json-objects
-	fmt.Println("Creating mapping for cities...")
-	mappingSql := fmt.Sprintf(`
-        CREATE OR REPLACE MAPPING city (
-			__key INT,
-			country VARCHAR,
-			city VARCHAR,
-			population BIGINT)
-        TYPE IMAP 
-        OPTIONS (
-            'keyFormat' = 'int',
-            'valueFormat' = 'json-flat'
-        )`)
-	_, err := sqlService.Execute(context.Background(), mappingSql)
-	if err != nil {
-		panic(err)
-	}
-	fmt.Println("Mapping for cities has been created")
-	fmt.Println("--------------------")
-}
+// CitySerializer serializes the City object
+type CitySerializer struct{}
 
-func populateCities(client *hazelcast.Client) {
-	// see: https://docs.hazelcast.com/hazelcast/5.1/data-structures/creating-a-map#writing-json-to-a-map
-	fmt.Println("Populating 'city' map with JSON values...")
-	ctx := context.Background()
-	cities, err := client.GetMap(ctx, "city")
-	if err != nil {
-		panic(err)
-	}
-	mustPut(cities.Put(ctx, 1, city("AU", "Canberra", 467_194)))
-	mustPut(cities.Put(ctx, 2, city("CZ", "Prague", 1_318_085)))
-	mustPut(cities.Put(ctx, 3, city("EN", "London", 9_540_576)))
-	mustPut(cities.Put(ctx, 4, city("US", "Washington, DC", 7_887_965)))
-	fmt.Println("The 'city' map has been populated.")
-	fmt.Println("--------------------")
+func (s CitySerializer) Type() reflect.Type {
+	return reflect.TypeOf(City{})
 }
 
-func selectCitiesByCountry(sqlService sql.Service, country string) {
-	sql := "SELECT city, population FROM city WHERE country=?"
-	fmt.Println("Select city and population with sql = " + sql)
-	result, err := sqlService.Execute(context.Background(), sql, country)
-	if err != nil {
-		fmt.Errorf("querying: %w", err)
-	}
-	defer result.Close()
-	iter, err := result.Iterator()
-	if err != nil {
-		fmt.Errorf("acquiring iterator: %w", err)
-	}
-	for iter.HasNext() {
-		row, err := iter.Next()
-		if err != nil {
-			fmt.Errorf("iterating: %w", err)
-		}
-		city := mustGet(row.GetByColumnName("city"))
-		population := mustGet(row.GetByColumnName("population"))
-		fmt.Printf("City = %s, Population = %d\n", city, population)
-	}
-	fmt.Println("--------------------")
+func (s CitySerializer) TypeName() string {
+	return "city"
```

Comment:
I would use something like: `github.com/hazelcast/hazelcast-cloud-go-sample/City` in order not to introduce any possible collisions.

Reply:
Well, I've used `city` intentionally as well as in other samples too in case of that someone would like to try multiple languages.